### ISSUE TITLE
GoalStep数据集label疑问

### ISSUE THREAD

**Issue by 97winwins** [Yisen-Feng/OSGNet#5]
您好，非常感谢您开源 OSGNet 代码和预处理数据！！！
我在使用 GoalStep 数据集进行训练时，遇到了一些关于 label 字段和 object feature 的疑问：
在 ego4d_multitask_loader.py 的注释里看到有提到：# narration 没有物体特征 ——> 可以理解为GoalStep 的 jsonl 文件里本身就没有 label 字段嘛？
在代码中，object feature 的相关函数（eg. _get_object_class_feat）需要依赖 video_item['label']。但在 GoalStep 数据集中，这个字段是缺失的，导致如果启用 object_feat 分支就会报错。
我的疑问是：
①在训练 GoalStep 时，是否本来就不启用 object_feat 分支？
②也就是说，GoalStep 的训练流程是不是应该只依赖 video feature + query feature + lavila caption / shot 分支，而 object feature 在这个任务中默认是关闭的？
我现在的做法是把 config 里相关参数（object_feat_dir、classname_feat_dir 等）设为 null，训练可以正常运行，但不确定这样是否符合您的原始设计。
想确认一下，这样理解是否正确？还是我遗漏了某些数据或处理步骤？
非常感谢！谢谢您！！！

**Comment by 97winwins** [comment]
> 我的[标注文件](https://github.com/Yisen-Feng/OSGNet/blob/main/ego4d_data/goalstep_data/ego4d_goal_step_test_v2_lemma.jsonl)有label标签的

好的感谢您的回答！！！非常感谢！！！根据您的路径，请问这个数据集是不是应该指向val_lemma.jsonl？

<img width="1260" height="476" alt="Image" src="https://github.com/user-attachments/assets/4808da30-6f5e-4e5f-9b28-70cd68cb8f1e" />

**Comment by 97winwins** [comment]
> 我在测试和验证时的输入长度是比训练时大的，没有去解决这个问题。
嗷嗷好的谢谢哥，那所有实验四张48G的足够吗？🤔我目前有的是4张A6000，按照您说的就把以下2个地方改一下就行了吗？（因为估算了一下感觉goalstep训练时间较长，所以想提前和您确认一下）
1、train.py里面的# model = DDP(model, device_ids=[gpu_id])注释取消改成“model = DDP(model, device_ids=[gpu_id], find_unused_parameters=True)，请问您在训练的时候采用4卡时是不是也将这一行取消了呀🫶
”
2、我发现train.py好像并不会自动跳成eval模式，一直处于train，这也就是之前，于是我加了这两处：model.train()、model.eval()，不知道行不行（真的很感谢哥解答我这些问题🥹）

<img width="711" height="484" alt="Image" src="https://github.com/user-attachments/assets/e3b38e01-1e38-4c59-922a-af616d129844" />

<img width="692" height="287" alt="Image" src="https://github.com/user-attachments/assets/779a9594-b642-4b0b-bf56-04f61772493f" />

非常感谢哥，论文对本小白来说收获很大！太负责了！！！

**Comment by Yisen-Feng** [comment]
1.没有。
2.函数内部应该有设置train和eval模式。

**Comment by Yisen-Feng** [comment]
你应该是去计算验证集的loss了，算loss是要train模式的，而且会限制长度。
 > 这是readme中原句：For GoalStep, mode should be not-eval-loss.
 > 参考：https://github.com/Yisen-Feng/OSGNet/blob/main/README.md?plain=1#L134-L137

**Comment by 97winwins** [comment]
> 你应该是去计算验证集的loss了，算loss是要train模式的，而且会限制长度。
> 
> > 这是readme中原句：For GoalStep, mode should be not-eval-loss.
> > 参考：https://github.com/Yisen-Feng/OSGNet/blob/main/README.md?plain=1#L134-L137

omg太感谢哥在节假日仍然解答我的疑问，我这就试试看看情况！您太负责了！！！祝哥paper多多！！！🥰🙇‍♂️
